README.md

```diff
@@ -134,7 +134,194 @@ TODO
 
 ## Wire vs Protobuf
 
-TODO
+From the [Protocol Buffers encoding guide](https://developers.google.com/protocol-buffers/docs/encoding):
+
+> As you know, a protocol buffer message is a series of key-value pairs. The
+> binary version of a message just uses the field's number as the key – the
+> name and declared type for each field can only be determined on the decoding
+> end by referencing the message type's definition (i.e. the .proto file).
+>
+> When a message is encoded, the keys and values are concatenated into a byte
+> stream. When the message is being decoded, the parser needs to be able to
+> skip fields that it doesn't recognize. This way, new fields can be added to a
+> message without breaking old programs that do not know about them. To this
+> end, the "key" for each pair in a wire-format message is actually two values
+> – the field number from your .proto file, plus a wire type that provides just
+> enough information to find the length of the following value.
+>
+> The available wire types are as follows:
+> 
+> Type | Meaning | Used For
+> ---- | ------- | --------
+> 0    | Varint  | int32, int64, uint32, uint64, sint32, sint64, bool, enum
+> 1    | 64-bit  | fixed64, sfixed64, double
+> 2    | Length-delimited | string, bytes, embedded messages, packed repeated fields
+> 3    | Start group | groups (deprecated)
+> 4    | End group | groups (deprecated)
+> 5    | 32-bit  | fixed32, sfixed32, float
+>
+> Each key in the streamed message is a varint with the value (field_number <<
+> 3) | wire_type – in other words, the last three bits of the number store the
+> wire type.
+
+In Wire, a "type" can be encoded by more than 3 bits, due to the introduction
+of a "List" type, which can hold other types recursively.  So collectively,
+lets call these bits the `<type-bytes>`.
+
+In Wire, "varint" means protobuf's "signed varint", and "uvarint" means "varint".
```

Comment:
In Wire, "varint" is the Protobuf equivalent of "signed varint" aka `sint32` and "uvarint" is the equivalent of "varint" aka `int32`